jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -137,7 +137,10 @@ private static JavaLayerConfigurations getForNonWarProject(
         layerBuilder.addFile(
             layerType,
             dependencyFile.toPath(),
-            dependenciesExtractionPath.resolve(dependencyFile.getName()));
+            dependenciesExtractionPath.resolve(
+                dependencyFile
+                    .getName()
+                    .replace(".jar", "-" + Files.size(dependencyFile.toPath()) + ".jar")));
```

Comment:
Hmm, do we want to be adding size to all of the dependency jar names or just the duplicate ones?

Reply:
If we add it to all of them, we can skip sorting to maintain reproducibility.

Reply:
`ReproducibleLayerBuilder` currently already sorts no matter what and we wouldn't need to do any backtracking - just append the size if the name is a name that has already been encountered.

Reply:
Ah, sorry. I see what you are saying: you don't want to sort for finding duplicates, right?

Reply:
Right, the reason we'd need to sort is if we rename only duplicates. e.g. if file 1, 2, and 3 have the same name, file 1 would keep its name while 2 and 3 are renamed, but we'd need to sort the files first to keep this reproducible so we always encounter file 1 first to keep its original name, and rename file 2 and 3.

Reply:
Hmm... I'm a bit indecisive. The simple renaming works, but it's seems worth trying to rename only when necessary.

Reply:
I'm looking into renaming with group id instead of file size

Reply:
Ah, interesting. I thought it's not possible to obtain a Maven coordinates group ID in Gradle.

Reply:
Yeah, if this is possible it doesn't look like it's going to be very clean.

Reply:
Would look at the maven or gradle war plugin implementations, since those should be doing something like that

Reply:
Not having any luck doing that either :confused: